statsd/worker.go

```diff
@@ -59,7 +71,7 @@ func (w *worker) processMetric(m metric) error {
 }
 
 func (w *worker) shouldSample(rate float64) bool {
-	if rate < 1 && rand.Float64() > rate {
+	if rate < 1 && w.random.Float64() > rate {
```

Comment:
@matthewdale @hush-hush Shouldn't we have a local lock here? From the [docs](https://golang.org/pkg/math/rand/):
> The default Source is safe for concurrent use by multiple goroutines, **but Sources created by NewSource are not.**

I think this is causing race conditions.

Reply:
You're correct. When `MutexMode` is enabled, the call to `worker.processMetric()` is made from the same goroutine as the call to the top-level metric emit function (e.g. `Count`). I was previously under the impression that all calls to `worker.processMetric()` would be made from a single goroutine per worker, but that's not the case.

I'm working on a fix for the data race bug and will submit it when it's ready.